src/Data/Array/Accelerate/AST/Idx.hs

```diff
@@ -17,19 +25,19 @@ module Data.Array.Accelerate.AST.Idx
   where
 
 import Language.Haskell.TH
+import Data.Type.Equality ((:~:)(Refl))
+import Unsafe.Coerce (unsafeCoerce)
```

Comment:
Do these imports generate "unused import" warnings when in internal-checks mode? If so, maybe put them inside an `#ifndef` in order to not introduce spurious warnings.

Reply:
They do, but I'm not sure whether some 'unused import' warnings on a rare internal flag are really worse than multiple CPP's that depend on each other? Maybe I'm too allergic to these #ifdef's

Reply:
@tmcdonell I believe other files do try to make sure that setting flags doesn't cause unused import warnings; your thoughts?